### PR TITLE
Feature 665/USFM linter

### PR DESCRIPTION
**Pull request by PhotoNomad0**
Issue: https://github.com/unfoldingWord-dev/door43.org/issues/665

Changes:
- Pulled in Bible classes from uw-publish to use in checker.  
- Renamed bible classes to be USFM classes.
- Tweaked code and regex to capture more errors.

### CODE REVIEW

**Comment by PhotoNomad0** [comment]
@richmahn - ready for review.  Think it is time to disable InteliJ warnings of camel case variable/method names, because I don't want the fix the case of several hundred methods in the included code.

File: libraries/checkers/usfm_checker.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+import os
+import traceback
+from libraries.checkers.checker import Checker
+from libraries.door43_tools.page_metrics import PageMetrics
+from libraries.usfm_tools import verifyUSFM, usfm_verses
+
+
+class UsfmChecker(Checker):
+
+    def __init__(self, preconvert_dir, converted_dir, log=None):
```
Comment (richmahn) [inline libraries/checkers/usfm_checker.py]: It's always best to not to specifically name variables used by the parent init, as those may change (which they have in my branch). Do this instead for the first two lines:
```
    def __init__(self, *args, **kwargs):
        super(UsfmChecker, self).__init__(*args, **kwargs)
```

File: libraries/checkers/usfm_checker.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        """
+
+        lang_code = self.rc.resource.language.identifier
+        valid_lang_code = PageMetrics().validate_language_code(lang_code)
+        if not valid_lang_code:
+            self.log.warning("Invalid language code: " + lang_code)
+
+        for root, dirs, files in os.walk(self.preconvert_dir):
+            for f in files:
+                if f[-3:].lower() != 'sfm':  # only usfm files
```
Comment (richmahn) [inline libraries/checkers/usfm_checker.py]: What I'd do here, as I do in markdown, is:

```if os.path.splitext(f)[1].lower() != '.usfm':```

splitext always give a two item list, with the second being the extension with a preceding '.'

File: libraries/checkers/usfm_checker.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                sub_path = '.' + file_path[len(self.preconvert_dir):]
+                self.parse_file(file_path, sub_path, f)
+
+        found_book_count = len(self.found_books)
+        if found_book_count == 0:
+            self.log.error("No translations found")
+        elif found_book_count == 1:
+            pass  # this is OK, presume this was a single book project
+        else:
+            self.warn_on_missing_books(self.found_books)
```
Comment (richmahn) [inline libraries/checkers/usfm_checker.py]: I don't find anywhere in the requirements to care if a whole book is missing, only its title is missing.

**Review by richmahn — CHANGES_REQUESTED** [review]
- A few code changes
- Does not match requirements - do not need to report missing books, only missing book titles.

File: libraries/checkers/usfm_checker.py (hunk 3 above)
Comment (jag3773) [reply libraries/checkers/usfm_checker.py]: Correct @richmahn, we will often have projects with multiple books but not all books.  This is OK.

**Review by richmahn — APPROVED** [review]
LGTM

**Pull request merged**